File: Stacks/isPalindrome.js
function Stack()  {
  this.dataStore = [];
  this.top =0;
  this.pop = pop;
  this.push = push;
  this.peek = peek;
  this.length= length;

  function push(element){
    this.dataStore[this.top++] = element;

  }

  function pop(){
    return this.dataStore[--this.top];
  }


  function peek(){
    return this.dataStore[this.top-1];
  }

  function length(){
    return this.top;
  }

  function clear(){
    this.top = 0;
  }

}


function isPalindrome(word){
       var s = new Stack();
      for (var i = 0; i < word.length; i++){
        s.push(word[i]);
      }
          var secWord = "";
          while(s.length() > 0){
            secWord += s.pop();
          }
          if (word == secWord){
            return true;
          }
          else{
            return false;
          }

}
console.log((isPalindrome("ROTAVATOR"))?(`Is a Palindrome`):(`Is not a Palindrome`));

File: Stacks/isPalindrome.cpp
#include <iostream>
#include <string>
//#include "Stack.h"
using namespace std;

struct Node
{
char data;
Node* link;
};
Node* createNode(char data,Node* link)
{
Node* tmp = new Node;
tmp->data = data;
tmp->link = link;
return tmp;
}

struct Stack
{
Node* top;
};

void initialize(Stack& _this)
{
_this.top = NULL;
}

bool isEmpty(const Stack& _this)
{
return _this.top == NULL;
}

char top(const Stack& _this)
{
return _this.top->data;
}

void push(Stack& _this,char item)
{
Node* tmp = createNode(item, _this.top);
_this.top = tmp;
}

void pop(Stack& _this)
{
if(!isEmpty( _this))
{
Node* tmp = _this.top;
_this.top = _this.top->link;
delete tmp;
tmp = NULL;
}
}




 bool IsPalindrome(char *str){
  Stack stack;
  initialize(stack);
  for(int i = 0; str[i] != '\0'; i++){
    push(stack, str[i]);
  }
  for(char *i = str ; (!isEmpty(stack)); *i++ )
    if (top(stack) == *i){
      pop(stack);
    }
    else if (top(stack) != *i){
      return false;
    }
  if(isEmpty(stack)){
    return true;
  }
  return false;
  }

int main()
{
  char *pointer_, array_[512];

  cout << "Enter A String:";
  cin >> array_;
  pointer_ = array_;
  if( IsPalindrome(pointer_)){
   cout << "true";
    }else{
   cout << "false";
    }
    return 0;
    }
